### ISSUE TITLE
PowerSaver & Wifi class cannot be assigned

### ISSUE THREAD

**Issue by stoffera** [getopenmono/mono_framework#23]
The assignment operator from #15 does not work

### Problem

The assignment operator for both classes does not define its parameter as `const`, which it needs to be. The operator right changes the object which is assigned, therefore it cannot be `const`.

### Solution

The operator overload must be `const`. This means the parameter (the object that gets assigned) cannot be changed. This means there need to exists two copies of the object, which can be problem for the Wifi class, since it owns an instance of `ModuleSPICommunication`

**Comment by jpsecher** [comment]
Maybe [`mutable`](https://stackoverflow.com/questions/105014/does-the-mutable-keyword-have-any-purpose-other-than-allowing-the-variable-to) will come in handy?

**Comment by stoffera** [comment]
@jpsecher So inside the `Wifi` class we mark its member `ModuleSPICommunication` as `mutable`- to allow it to change - even if the owning object is `const`?

**Comment by jpsecher** [comment]
Yes, although I don't know if that is a good idea in this particular case, haven't looked at the code...
